src/Microsoft.AspNet.Mvc.Core/Formatters/MediaTypeComparisons.cs

```diff
@@ -0,0 +1,292 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNet.Mvc.Formatters
+{
+    public static class MediaTypeComparisons
+    {
+        /// <summary>
+        /// Determines if the <paramref name="subset" /> media type is a subset of the <paramref name="set" /> media type
+        /// without taking into account the quality parameter.
+        /// </summary>
+        /// <param name="set">The more general media type.</param>
+        /// <param name="subset">The more specific media type.</param>
+        /// <returns><code>true</code> if <paramref name="set" />is a more general media type than <paramref name="subset"/>
+        /// except for the quality parameter; otherwise <code>false</code>.</returns>
+        public static bool IsSubsetOf(StringSegment set, string subset)
+        {
+            return IsSubsetOf(set, new StringSegment(subset), ignoreQuality: true);
+        }
+
+        /// <summary>
+        /// Determines if the <paramref name="subset" /> media type is a subset of the <paramref name="set" /> media type
+        /// without taking into account the quality parameter.
+        /// </summary>
+        /// <param name="set">The more general media type.</param>
+        /// <param name="subset">The more specific media type.</param>
+        /// <returns><code>true</code> if <paramref name="set" />is a more general media type than <paramref name="subset"/>
+        /// except for the quality parameter; otherwise <code>false</code>.</returns>
+        public static bool IsSubsetOf(string set, string subset)
+        {
+            return IsSubsetOf(new StringSegment(set), new StringSegment(subset), ignoreQuality: true);
+        }
+
+        /// <summary>
+        /// Determines if the <paramref name="subset" /> media type is a subset of the <paramref name="set" /> media type.
+        /// Two media types are compatible if one is a subset of the other ignoring any charset
+        /// parameter.
+        /// </summary>
+        /// <param name="set">The more general media type.</param>
+        /// <param name="subset">The more specific media type.</param>
+        /// <param name="ignoreQuality">Whether or not we should skip checking the quality parameter.</param>
+        /// <returns><code>true</code> if <paramref name="set" />is a more general media type than <paramref name="subset"/>;
+        /// otherwise <code>false</code>.</returns>
+        public static bool IsSubsetOf(StringSegment set, StringSegment subset, bool ignoreQuality)
```

Comment:
@javiercn +1 to what @rynowak said. You're porting `MediaTypeHeaderValue` code with the signature `public bool IsSubsetOf(MediaTypeHeaderValue otherMediaType)`. There's no `ignoreQuality` equivalent -- effectively it's always `true`. So what would call this new method with `ignoreQuality: false`?

Separately you aren't correctly ignoring the `q` parameter. `IsSubsetOf()` should ignore the `q` parameter and _all following parameters_. See [this block](https://github.com/aspnet/HttpAbstractions/blob/dev/src/Microsoft.Net.Http.Headers/MediaTypeHeaderValue.cs#L288-L293).